frontend/src/app/pages/login/login.component.ts

```diff
@@ -1,10 +1,103 @@
-import { Component } from '@angular/core';
+import { Component, NgZone, OnDestroy, OnInit } from '@angular/core';
+import { AbstractControl, FormControl, FormGroup, Validators } from '@angular/forms';
+import { MessageService } from 'primeng/api';
+import { finalize, Observable, of, Subscription, switchMap, tap } from 'rxjs';
+import { Router } from '@angular/router';
+import { UserForLoginDTO } from 'src/app/authentication/models/user-login-dto';
+import { UserService } from 'src/app/core/services/user.service';
+import { AuthenticationResponseDTO } from 'src/app/authentication/models/authentication-response-dto';
 
 @Component({
   selector: 'app-login',
   templateUrl: './login.component.html',
   styleUrls: ['./login.component.scss']
 })
-export class LoginComponent {
+export class LoginComponent implements OnDestroy {
+    loginForm: FormGroup = new FormGroup({
+    email: new FormControl('', [Validators.required, Validators.email]),
+    password: new FormControl('', Validators.required)
+  });
+  subscriptions: Subscription[] = [];
+  isProgressSpinnerVisible = false;
+  passwordInputType = 'password';
+  passwordEyeIcon = 'pi-eye';
 
-}
+  constructor(
+    private userService: UserService,
+    private messageService: MessageService,
+    private router: Router) { }
+
+  ngOnDestroy(): void {
+    this.subscriptions.forEach(subscription => {
+      subscription.unsubscribe();
+    });
+  }
+
+  onSubmit(): void {
+    if (this.loginForm.invalid) {
+      this.loginForm.markAllAsTouched();
+      return;
+    }
+
+    const userForLogin: UserForLoginDTO = {
+      email: this.loginForm.get('email')!.value,
+      password: this.loginForm.get('password')!.value
+    };
+
+    const login$ = this.userService.loginUser(userForLogin).pipe(
+      tap(() => {
```

Comment:
if we are getting navigated to home page maybe we shouldnt display a message

Reply:
I don't get your obsessionwith displaying messages